src/App.tsx

```diff
@@ -397,9 +402,21 @@ function App(): ReactElement {
     });
   };
 
+  const getFeatureDropdownData = useCallback((): string[] | { key: string; label: string }[] => {
+    if (!dataset) {
+      return [];
+    }
+    // Add units to the dataset feature names if present
+    return dataset.featureNames.map((name) => {
+      return { key: name, label: dataset.getFeatureNameWithUnits(name) };
+    });
+  }, [dataset]);
+
   const disableUi: boolean = isRecording || !datasetOpen;
   const disableTimeControlsUi = disableUi;
 
+  const featureUnits = dataset?.featureHasUnits(featureName) ? dataset.getFeatureUnits(featureName) : "";
```

Comment:
Nit: getFeatureUnits could do the featureHasUnits check internally and return empty string as needed...?

Reply:
Maybe horrid idea, what if someone includes empty quotes for their unit string? Should we allow that...?

Reply:
If they actually put a string consisting of two consecutive quotation marks (`"\"\""`), then we should print it as such.  But if they specify empty string, then that's the same as "no units provided".